.github/workflows/release-main.yml

```
@@ -37,6 +37,11 @@ jobs:
         dotnet-sdk-version: ${{ env.DOTNET_VERSION }}
         command: ./build.sh generateapichanges -s true
 
+    - name: publish canary packages github package repository
+      shell: bash
```

What about a timeout if taking longer than X minutes?

```suggestion
      shell: bash
      timeout-minutes: 10
```

OTOH, will this step need to run before the other ones? If it fails, should fail always?